internal/statusapi/statusapi.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package statusapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/newrelic/infrastructure-agent/internal/agent/status"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+)
+
+const (
+	IntegrationName         = "status-api"
+	statusAPIPath           = "/v1/status"
+	statusOnlyErrorsAPIPath = "/v1/status/errors"
+)
+
+type responseError struct {
+	Error string `json:"error"`
+}
+
+// Server runtime for status API server.
+type Server struct {
+	host     string
+	port     int
+	reporter status.Reporter
+	logger   log.Entry
+	readyCh  chan struct{}
+}
+
+// NewServer creates a new status local-only, read-only, API server.
+func NewServer(port int, r status.Reporter) *Server {
+	return &Server{
+		host:     "localhost", // local only API
+		port:     port,
+		reporter: r,
+		logger:   log.WithComponent(IntegrationName).WithField("port", port),
+		readyCh:  make(chan struct{}),
+	}
+}
+
+// Serve serves status API requests.
+func (s *Server) Serve(ctx context.Context) {
+	router := httprouter.New()
+	// read only API
+	router.GET(statusAPIPath, s.handle(false))
```

Comment:
handle function will be blocked if the agent is not able to do the connect call?

Reply:
nope, id.Provide receives the non-blocking provide ID or empty in case there's no value yet